src/components/ec-full-screen-overlay/ec-full-screen-overlay.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <transition name="ec-full-screen-overlay__fade">
+    <div
+      v-if="show"
+      class="ec-full-screen-overlay"
+      data-test="ec-full-screen-overlay"
+    >
+      <div
+        ref="overlayContent"
+        class="ec-full-screen-overlay__content"
+        data-test="ec-full-screen-overlay__content"
+      >
+        <header
+          class="ec-full-screen-overlay__header"
+          data-test="ec-full-screen-overlay__header"
+        >
+          <h1
+            v-if="title"
+            data-test="ec-full-screen-overlay__title"
+            class="ec-full-screen-overlay__title"
+          >
+            {{ title }}
+          </h1>
+
+          <button
+            class="ec-full-screen-overlay__close-icon-btn"
+            data-test="ec-full-screen-overlay__close-icon-btn"
+            @click="emit('close-overlay')"
+          >
+            <span class="tw-sr-only">Close</span>
+            <ec-icon
+              class="ec-full-screen-overlay__close-icon"
+              data-test="ec-full-screen-overlay__close-icon"
+              name="simple-close"
+              type="interactive"
+              :size="24"
+            />
+          </button>
+
+        </header>
+        <main data-test="ec-full-screen-overlay__main">
+          <slot name="main" />
+        </main>
+      </div>
+
+    </div>
+  </transition>
+</template>
+
+<script setup>
+import { useFocusTrap } from '@vueuse/integrations/useFocusTrap';
+import { onUnmounted, ref } from 'vue';
+
+import EcIcon from '../ec-icon';
+
+defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  show: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const emit = defineEmits(['close-overlay']);
+
+const overlayContent = ref(null);
+const { deactivate } = useFocusTrap(overlayContent, {
+  immediate: true,
+  escapeDeactivates: false,
+  clickOutsideDeactivates: false,
+  initialFocus: false,
+});
+
+onUnmounted(() => {
+  deactivate();
+});
+
+</script>
+
+<style>
+  @import '../../styles/tools/transitions.css';
+  @import '../../styles/tools/ec-full-screen-overlay.css';
+
+  .ec-full-screen-overlay {
+    @apply tw-min-h-screen;
+    @apply tw-z-level-3;
+    @apply tw-fixed;
+    @apply tw-inset-0;
+    @apply tw-bg-gray-8;
+    @apply tw-flex;
+    @apply tw-justify-center;
+    @apply tw-overflow-y-auto;
+
+    &__content {
+      max-width: 704px;
+      width: 704px;
```

Comment:
Few notes on this one:
* potentially this could have been 100% so we will not have to change 2 values when we change the width.
* Did designers confirm the 704px that is the correct value?
* Do we want this at the end with adjustable width as a prop or we hard code for now?

Reply:
Use a CSS variable so it can we overridden if needed in EBO.